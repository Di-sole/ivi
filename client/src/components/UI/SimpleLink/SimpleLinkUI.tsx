import Link from 'next/link';
import styles from './SimpleLinkUI.module.scss';

interface SimpleLinkUIProps {
  href: string;
  children: React.ReactNode;
  option: 'bright' | 'dim';
}

const SimpleLinkUI = ({ href, children, option }: SimpleLinkUIProps) => {
  return (
    <Link href={href} className={[styles.link, styles[`link_${option}`]].join(' ')}>
      {children}
    </Link>
  );
};

export default SimpleLinkUI;
