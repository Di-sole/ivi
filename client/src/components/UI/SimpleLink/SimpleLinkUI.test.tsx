import { render, screen } from '@testing-library/react';
import SimpleLinkUI from './SimpleLinkUI';

describe('SIMPLELINKUI SNAPSHOTS TESTS', () => {
  test('bright SIMPLELINKUI should not be changed', () => {
    render(
      <SimpleLinkUI href="/" option="bright">
        Ссылка
      </SimpleLinkUI>,
    );
    const link = screen.getByRole('link');
    expect(link).toMatchSnapshot();
  });

  test('dim SIMPLELINKUI should not be changed', () => {
    render(
      <SimpleLinkUI href="/" option="dim">
        Ссылка
      </SimpleLinkUI>,
    );
    const link = screen.getByRole('link');
    expect(link).toMatchSnapshot();
  });
});
