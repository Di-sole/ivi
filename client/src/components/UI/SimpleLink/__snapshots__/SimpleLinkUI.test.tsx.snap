// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIMPLELINKUI SNAPSHOTS TESTS bright SIMPLELINKUI should not be changed 1`] = `
<a
  class="link link_bright"
  href="/"
>
  Ссылка
</a>
`;

exports[`SIMPLELINKUI SNAPSHOTS TESTS dim SIMPLELINKUI should not be changed 1`] = `
<a
  class="link link_dim"
  href="/"
>
  Ссылка
</a>
`;
